codec-http2/src/test/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionEncoderTest.java

```diff
@@ -200,20 +201,6 @@ public Object answer(InvocationOnMock invocationOnMock) throws Throwable {
     }
 
     @Test
-    public void dataWriteAfterGoAwayShouldFail() throws Exception {
```

Comment:
> Personally, I think the only classes that in the HTTP/2 tests should be mocked is a Channel, ChannelHandlerContext and the FrameWriter

I don't necessarily disagree with you and the "new" tests I have contributed are at a "higher" level (don't mock everything around them).  One thing I can say for mocks is that they can make the following easier...
1. Simulating "extraneous to what is under test" interfaces and state.
2. Verifying state and interface interaction.

Sometimes with full up implementations this is difficult to do at the granularity you need (depending on design, etc...).

> So I added those four tests 

Look good :) Thanks!